### PR TITLE
Replaced call to Arrangement_on_surface_2::traits() with geometry_traits() and cleaned up

### PR DESCRIPTION
**Pull request by efifogel**
This is a small fix that replaces the call to the member traits() of Arrangement_on_surface_2 with the call geometry_traits(). This is another step in utilizing the trapezoidal decomposition point-location location strategy for arrangements on spheres.

## Release Management

* Affected package(s): Arrangement_on_surface_2
* Issue(s) solved (if any): NA
* Feature/Small Feature (if any): NA
* Link to compiled documentation: NA
* License and copyright ownership: TA univ

### CODE REVIEW

File: Arrangement_on_surface_2/include/CGAL/Arr_point_location/Arr_trapezoid_ric_pl_impl.h [hunk 1]
```diff
@@ -316,48 +316,41 @@ _vertical_ray_shoot(const Point_2& p, bool shoot_up) const
...
+  Vertex_const_handle closest_iso_v;
+  const Vertex_const_handle invalid_v;
+  const Halfedge_const_handle invalid_he;
 
   // If the closest feature is a valid halfedge, take its incident face.
   // Otherwise, take the unbounded face.
-  if (halfedge_found == invalid_he)
-    face = _get_unbounded_face(tr, p, All_sides_oblivious_category());
-  else
+  Face_const_handle face = (halfedge_found == invalid_he) ?
```
Comment (sloriot) [inline Arrangement_on_surface_2/include/CGAL/Arr_point_location/Arr_trapezoid_ric_pl_impl.h:339]: This triggers a new warning:
`Arrangement_on_surface_2/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Arr_point_location/Arr_trapezoid_ric_pl_impl.h:339:21: warning: operation on 'face' may be undefined [-Wsequence-point]
`

**Comment by efifogel** [comment]
ok, this is fixed now; thanks.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by efifogel** [comment]
Added back an explicit definition of an isolated-vertex iterator (without
'auto') to pacify MSVC.
Please try again.
Thanks.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-99

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-119

**Pull request merged**